src/new-components/checkbox-field/checkbox-field.test.tsx

```diff
@@ -54,11 +54,22 @@ describe('CheckboxField', () => {
         expect(screen.getByTestId('checkbox-field')).toHaveAccessibleName('Show unread badge')
     })
 
-    it('is labelled by its label', () => {
+    it('is labelled by its `String` label', () => {
         render(<CheckboxField data-testid="checkbox-field" label="Show completed tasks" />)
         expect(screen.getByTestId('checkbox-field')).toHaveAccessibleName('Show completed tasks')
     })
 
+    it('is labelled by its `ReactNode` label', () => {
+        render(
+            <CheckboxField
+                data-testid="checkbox-field"
+                label={<span data-testid="react-node-label">Show completed tasks</span>}
+            />,
+        )
+        expect(screen.getByTestId('checkbox-field')).toHaveAccessibleName('Show completed tasks')
+        expect(screen.getByTestId('react-node-label')).toHaveTextContent('Show completed tasks')
```

Comment:
This line is not testing anything about the library. It is testing something you rendered in the test.

That is, there's nothing you can change in the component code that would make this assertion fail.

I suggest we remove it:

```suggestion
```

Reply:
I understand what you mean, but I added this as a way to test that you can pass a `ReactNode` to the component, and that it's rendered properly. This test correctly asserts for that. I feel that removing this assertion, renders the new test I added irrelevant, don't you think?

Reply:
If you can make a single change in the component code that breaks this assertion, and this assertion only, I'll agree with you.

Reply:
If what you want to test is about making sure that the react node content is rendered verbatim, I'd say there are clearer ways to do that. For instance, use a `label={<>Hello <strong>world</strong></>}` and then assert:

```js
expect(screen.getByText('world').tagName).toBe('strong')
```

Or maybe you can do:

```suggestion
        expect(screen.getByTestId('react-node-label')).toBeInTheDocument()
```

No need to test the text content twice. I think that's what tipped me off a bit.

Reply:
> No need to test the text content twice. I think that's what tipped me off a bit.

In the end, all 3 alternatives will assert the same thing, I'm okay with switching to one of those simpler options 👍